frontend/src/metabase/query_builder/components/EditableText/EditableText.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from "react";
+
+import { Root, StyledTextArea } from "./EditableText.styled";
+
+import { KEY_ESCAPE } from "metabase/lib/keyboard";
+
+type text = string | null | undefined;
+
+interface Props {
+  value: text;
+  onChange?: (val: string) => void;
+}
+
+const EditableText = ({ value, onChange }: Props) => {
+  const [memo, setMemo] = useState<text>(value);
+
+  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setMemo(e.target.value);
+  };
+
+  const handleBlur = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    if (onChange) {
+      onChange(e.target.value);
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === KEY_ESCAPE) {
+      setMemo(value);
```

Comment:
There is an issue that by editing, cancelling, and editing again we would reset to an incorrect value here. Maybe we should store the latest "committed" value separately